sat/cli/setrev/main.py

```diff
@@ -54,18 +53,23 @@ def get_site_data(sitefile):
     Raises:
         PermissionError: If the sitefile exists locally but could not be read.
     """
-    s3 = get_s3_resource()
-    s3_bucket = get_config_value('s3.bucket')
-
     try:
-        LOGGER.debug('Downloading %s from S3 (bucket: %s)', sitefile, s3_bucket)
-        s3.Object(s3_bucket, sitefile).download_file(sitefile)
-    except (BotoCoreError, ClientError, Boto3Error) as err:
-        # It is not an error if this file doesn't already exist
-        # TODO: it would be nice to differentiate between successfully connecting to S3
-        # and the file not existing versus failing to connect to S3, where the latter
-        # deserves at least a warning-level message.
-        LOGGER.debug('Unable to download site info file %s from S3: %s', sitefile, err)
+        s3 = get_s3_resource()
+    except S3ResourceCreationError as err:
+        LOGGER.error('Error creating S3 resource: %s', err)
+        s3 = None  # Set s3 to None to skip S3 operations
+
+    s3_bucket = get_config_value('s3.bucket')
+    if s3:
+        try:
+            LOGGER.debug('Downloading %s from S3 (bucket: %s)', sitefile, s3_bucket)
+            s3.Object(s3_bucket, sitefile).download_file(sitefile)
+        except (BotoCoreError, ClientError, Boto3Error) as err:
+            # It is not an error if this file doesn't already exist
+            # TODO: it would be nice to differentiate between successfully connecting to S3
+            # and the file not existing versus failing to connect to S3, where the latter
+            # deserves at least a warning-level message.
+            LOGGER.debug('Unable to download site info file %s from S3: %s', sitefile, err)
```

Comment:
You could just use an `else` on the try/except like this:

```suggestion
    else:
        s3_bucket = get_config_value('s3.bucket')
        try:
            LOGGER.debug('Downloading %s from S3 (bucket: %s)', sitefile, s3_bucket)
            s3.Object(s3_bucket, sitefile).download_file(sitefile)
        except (BotoCoreError, ClientError, Boto3Error) as err:
            # It is not an error if this file doesn't already exist
            # TODO: it would be nice to differentiate between successfully connecting to S3
            # and the file not existing versus failing to connect to S3, where the latter
            # deserves at least a warning-level message.
            LOGGER.debug('Unable to download site info file %s from S3: %s', sitefile, err)
```